pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -119,3 +120,13 @@ func (pl *PodTopologySpread) setListers(factory informers.SharedInformerFactory)
 	pl.replicaSets = factory.Apps().V1().ReplicaSets().Lister()
 	pl.statefulSets = factory.Apps().V1().StatefulSets().Lister()
 }
+
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *PodTopologySpread) EventsToRegister() []framework.ClusterEvent {
+	return []framework.ClusterEvent{
+		// pod may be fail by new pod created or pod delete in some node with specified label
+		// And updated node may make pods schedulable.
+		{Resource: framework.Pod, ActionType: framework.Add | framework.Delete | framework.Update},
```

Comment:
new nodes could also make a pod schedulable right?

Reply:
Yes, NodeAdd/NodeDelete/NodeUpdateLable will re-evaluate the Pods spreading, and the unschedulable Pod may become schedulable:

```go
{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.UpdateNodeLabel},
```

Reply:
New node add will with label seted?

Reply:
Node delete will with label unset, this will affect scheduling.